### DISCUSSION TITLE
Does Co-Pilot provides supports for TCL Language?

### DISCUSSION THREAD

**Discussion by sanjaymishraGit** [community/community#118514] [Copilot Conversations]
### Select Topic Area

General

### Body

Does Co-Pilot has support for TCL Language ? Do we gat some simple suggestions or features like linting, style guide for TCL language by using Co-pilot ?

**Comment by jfullstackdev** [comment]
and from the answer of Copilot

---

As an AI developed by OpenAI and GitHub, GitHub Copilot has been trained on a wide variety of programming languages, including TCL. It can provide suggestions for writing TCL code based on the context provided by the user.

However, GitHub Copilot is primarily a code completion tool and does not directly provide features like linting or enforcing a style guide. These features are typically provided by other tools or extensions specific to the programming language.

For TCL, you might want to look into tools like `tclchecker` for linting, or follow a style guide such as the one provided by Tcl Developer Xchange. You can use these tools in conjunction with GitHub Copilot to write clean and effective TCL code.

---

as for me, I use SonarLint but this is more generic but covers most languages rules, but of course if you know one linting, then stick to it, as for style checking, for example, I use PHP CS Fixer for Laravel as it is in PHP together with SonarLint
